ecl/hthor/hthor.cpp

```diff
@@ -2494,14 +2494,62 @@ const void *CHThorGroupDedupKeepRightActivity::nextRow()
 
 //=====================================================================================================
 
+CHThorGroupDedupKeepBestActivity::CHThorGroupDedupKeepBestActivity(IAgentContext &_agent, unsigned _activityId, unsigned _subgraphId, IHThorDedupArg &_arg, ThorActivityKind _kind, bool _keepLeft)
+    : CHThorGroupDedupActivity(_agent, _activityId, _subgraphId, _arg, _kind), keepLeft (_keepLeft)
+{
+    compareBest = helper.queryCompareBest();
+}
+
+void CHThorGroupDedupKeepBestActivity::ready()
+{
+    CHThorGroupDedupActivity::ready();
+    assertex(numToKeep==1);
+    firstDone = false;
+}
+
+void CHThorGroupDedupKeepBestActivity::stop()
+{
+    kept.clear();
+    CHThorGroupDedupActivity::stop();
+}
+
+const void *CHThorGroupDedupKeepBestActivity::nextRow()
+{
+    if (!firstDone)
+    {
+        firstDone = true;
+        kept.setown(input->nextRow());
+    }
+
+    OwnedConstRoxieRow next;
+    loop
+    {
+        next.setown(input->nextRow());
+        if (!kept || !next || !helper.matches(kept,next))
+            break;
+
+        int cmp = compareBest->docompare(kept,next);
```

Comment:
A created a seperate activity to minimise the number of comparisions, so performance is a little better.  I see that KeepLeft and KeepRight were also separated into separate activities. If we do merge them into one of the existing activities, we can expect the existing activity to be slightly slower.. is this something we should do?